### DISCUSSION TITLE
Why do you use fetchQuery in router and not loadQuery?

### DISCUSSION THREAD

**Discussion by Dinno** [kriasoft/graphql-starter-kit#347] [Q&A]
This might be a naive question. I just begin learning Relay. But it is not obvious why you don't load query and pass it's reference to the page but instead fetch and pass it's data directly.

**Comment by Dinno** [comment]
Perhaps it is connected to using it for SSR?

**Reply by koistya** [reply]
Yes, in order to being able access that data during full or partial (just <head> section) SSR, see `web/workers/proxy.ts` -- a Cloudflare Worker script that resolves a URL path to a certain route, e.g. `cosnt route = await resolveRoute({ path: "/users/john" })` and then injects page title, meta tags into the HTML page using that response from the router, containing `fetchQuery` payload. And also injects the serialized Relay store into the HTML `<script id="data" type="application/json">...</script>` so that when this page is being rendered in the browser, it doesn't have to maky any HTTP calls to the API during the initial (first) render.

**Reply by koistya** [reply]
BTW, it is possible to tweak the router it to either use `fetchQuery` or `loadQuery` depending on whether you need that data during server-side rendering or not.
